implementation/clabbert/src/main/java/pwr/jakprzyjade/clabbert/api/middlewares/ApiExceptionHandler.java

```diff
@@ -0,0 +1,63 @@
+/* @jakubzehner (C) 2024 */
+package pwr.jakprzyjade.clabbert.api.middlewares;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.NoHandlerFoundException;
+import pwr.jakprzyjade.clabbert.api.contracts.ErrorDto;
+import pwr.jakprzyjade.clabbert.domain.exceptions.AbstractException;
+
+@ControllerAdvice
+public class ApiExceptionHandler {
+
+    @ExceptionHandler(AbstractException.class)
+    public ResponseEntity<ErrorDto> handleAbstractException(
+            AbstractException exception, WebRequest request) {
+        var kind =
+                exception
+                        .getClass()
+                        .getSimpleName()
+                        .replaceAll("([a-z])([A-Z])", "$1-$2")
+                        .toLowerCase();
```

Comment:
**thought:** Scary. Wouldn't it also keep the `Exception` suffix? I prefered `account-not-found` to `account-not-found-exception` since the last part is redundant, but I'm kinda fine with both.

Reply:
Correct, but there will be no `-exception` if there is no word “Exception” in the exception name. 
![image](https://github.com/user-attachments/assets/603ca94b-a82e-48a0-b5ec-fe408ca40a53)


Although, in retrospect, they should include it in the name, and I'll change it so that it removes it from the exception name when converting to `kind`.